contracts/adapters/Managing.sol

```diff
@@ -83,11 +97,18 @@ contract ManagingContract is IManaging, DaoConstants, MemberGuard {
         );
     }
 
+    /**
+     * @notice Sponsor a proposal if the proposal id exists.
+     * @dev Only members are allowed to sponsor proposals.
+     * @param dao The dao address.
+     * @param proposalId The guild kick proposal id.
+     * @param data Additional data that can be used for offchain voting validation.
+     */
     function sponsorProposal(
         DaoRegistry dao,
         bytes32 proposalId,
         bytes calldata data
-    ) external override {
+    ) external override onlyMember(dao) {
```

Comment:
we don't need onlyMember here. The solution is usually to use onlyMember2(dao, sponsoredBy) in _sponsorProposal(...)